akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -599,6 +599,17 @@ object TestSubscriber {
       self
     }
 
+    /**
+     * Fluent DSL
+     *
+     * Same as `expectNoMessage(remaining)`, but correctly treating the timeFactor.
+     * NOTE! Timeout value is automatically multiplied by timeFactor.
+     */
+    def expectNoMessage(): Self = {
+      probe.expectNoMessage()
+      self
+    }
```

Comment:
Hmm, I would have thought we left this one out on purpose?

Reply:
Alright, thanks for clarifying.

Reply:
Anytime!

Reply:
but `expectNoMessage` is never what you want. It defaults to 3 seconds delays, making the tests very slow without any additional value (in most cases).

In Akka Typed we introduced a separate setting for the default timeout for `expectNoMessage` and that is much better, but in untyped there is no such thing.

I can't understand why we added it in TestKit when renamed from expectNoMsg to expectNoMessage. That would have been a good time to change the behavior.

See also https://github.com/akka/akka/blob/master/CONTRIBUTING.md#preferred-ways-to-use-timeouts-in-tests

Reply:
I agree with you on the fact that `expectNoMessage()` is never what one wants. It's not that I desperately need this method on my code, but while cleaning some warnings on Akka HTTP I found out there was a similar parameterless method on the testkit and I wanted to deprecate. During the PR discussion it came out that probably would be good to keep that overloaded method and add the missing corresponding one on the Streams testkit. Please find the relevant discussion here: https://github.com/akka/akka-http/pull/2292#discussion_r239473817


PS: I'm completely happy to close this PR and go back to Akka HTTP and deprecate the similar method.

Reply:
I leave it up to you to decide for Akka HTTP, but it can be implemented from there using probe.expectNoMessage(probe.remainingOrDefault)

Reply:
Sorry for the delay in replying.
Then I would close this PR and go over Akka HTTP and use the method you mention. What do you think @jrudolph and @raboof ?